src/main/java/org/opensearch/knn/index/KNNWeight.java

```diff
@@ -77,114 +73,118 @@ public Explanation explain(LeafReaderContext context, int doc) {
-            } else {
-                String engineName = fieldInfo.attributes().getOrDefault(KNN_ENGINE, KNNEngine.NMSLIB.getName());
-                knnEngine = KNNEngine.getEngine(engineName);
-                String spaceTypeName = fieldInfo.attributes().getOrDefault(SPACE_TYPE, SpaceType.L2.getValue());
-                spaceType = SpaceType.getSpace(spaceTypeName);
-            }
-
-            /*
-             * In case of compound file, extension would be <engine-extension> + c otherwise <engine-extension>
-             */
-            String engineExtension = reader.getSegmentInfo().info.getUseCompoundFile()
-                    ? knnEngine.getExtension() + KNNConstants.COMPOUND_EXTENSION : knnEngine.getExtension();
-            String engineSuffix = knnQuery.getField() + engineExtension;
-            List<String> engineFiles = reader.getSegmentInfo().files().stream()
-                    .filter(fileName -> fileName.endsWith(engineSuffix))
-                    .collect(Collectors.toList());
-
-            if(engineFiles.isEmpty()) {
-                logger.debug("[KNN] No engine index found for field {} for segment {}",
-                        knnQuery.getField(), reader.getSegmentName());
-                return null;
-            }
-
-            Path indexPath = PathUtils.get(directory, engineFiles.get(0));
-            final KNNQueryResult[] results;
-            KNNCounter.GRAPH_QUERY_REQUESTS.increment();
-
-            // We need to first get index allocation
-            NativeMemoryAllocation indexAllocation;
-            try {
-                indexAllocation = nativeMemoryCacheManager.get(
-                        new NativeMemoryEntryContext.IndexEntryContext(
-                                indexPath.toString(),
-                                NativeMemoryLoadStrategy.IndexLoadStrategy.getInstance(),
-                                getParametersAtLoading(spaceType, knnEngine, knnQuery.getIndexName()),
-                                knnQuery.getIndexName()
-                        ), true);
-            } catch (ExecutionException e) {
-                GRAPH_QUERY_ERRORS.increment();
-                throw new RuntimeException(e);
-            }
-
-            // Now that we have the allocation, we need to readLock it
-            indexAllocation.readLock();
-
-            try {
-                if (indexAllocation.isClosed()) {
-                    throw new RuntimeException("Index has already been closed");
-                }
-
-                results = JNIService.queryIndex(indexAllocation.getMemoryAddress(), knnQuery.getQueryVector(), knnQuery.getK(), knnEngine.getName());
-            } catch (Exception e) {
-                GRAPH_QUERY_ERRORS.increment();
-                throw new RuntimeException(e);
-            } finally {
-                indexAllocation.readUnlock();
+        SegmentReader reader = (SegmentReader) FilterLeafReader.unwrap(context.reader());
```

Comment:
It's an old code, probably formatted by spotless. But I agree that the check is missing. Let me add it in next commit

Reply:
I would say dont add it in this PR. We can fix later. This was changed due to spotless.

Reply:
@martin-gaievski I agree to @jmazanec15's comment. Shall we make one huge ( with caution ) spotless apply PR, so that it will not be a problem every-time we touch the code.

Reply:
Ok, let's leave it and address in later PR. Please approve @[VijayanB](https://github.com/VijayanB) if that was the only comment from your side.

Reply:
@VijayanB I set https://github.com/opensearch-project/k-NN/blob/main/gradle/formatting.gradle#L10 the ratchet from setting to apply spotless incrementally to avoid messing us the git history with a huge commit that changes everything. I have been thinking about removing this and doing a big sptless apply. We can probably discuss in an issue I can create.